src/renderer/services/file-management-system/test/file-management-system.test.ts

```diff
@@ -164,6 +156,97 @@ describe("FileManagementSystem", () => {
       expect(inFlightFssRequests).to.be.equal(0);
     });
 
+    it("restarts job WORKING in FSS", async () => {
+      // Arrange
+      const uploadId = "elephant091234124";
+      const { mtime: fileLastModified } =
+          await fs.promises.stat(testFilePath);
+      const fileLastModifiedInMs = fileLastModified.getTime();
+      const uploadJob: UploadJob = {
+        ...mockJob,
+        serviceFields: {
+          files: [
+            {
+              file: {
+                fileType: "text",
+                originalPath: testFilePath,
+              },
+            },
+          ],
+          type: "upload",
+        },
+      };
+      const workingUploadJob: UploadJob = {
+        ...mockJob,
+        status: JSSJobStatus.WORKING,
+        serviceFields: {
+          fssUploadId: uploadId,
+          lastModifiedInMS: fileLastModifiedInMs,
+          files: [
+            {
+              file: {
+                fileType: "text",
+                originalPath: testFilePath,
+              },
+            },
+          ],
+          type: "upload",
+        },
+      };
+      const completeUploadJob: UploadJob = {
+        ...mockJob,
+        status: JSSJobStatus.SUCCEEDED,
+        serviceFields: {
+          fssUploadId: uploadId,
+          lastModifiedInMS: fileLastModifiedInMs,
+          files: [
+            {
+              file: {
+                fileType: "text",
+                originalPath: testFilePath,
+              },
+            },
+          ],
+          type: "upload",
+        },
+      };
+      fss.fileExistsByNameAndSize.resolves(false);
+      fss.registerUpload.onFirstCall().resolves({ status: UploadStatus.WORKING, chunkStatuses: [UploadStatus.WORKING], uploadId, chunkSize: 2424 })
+          .onSecondCall().resolves({ status: UploadStatus.WORKING, chunkStatuses: [UploadStatus.WORKING,UploadStatus.WORKING], uploadId, chunkSize: 2424 });
+      jss.updateJob.resolves();
+      jss.createJob.resolves(workingUploadJob);
+      jss.getJob
+          .onFirstCall().resolves(uploadJob)
+          .onSecondCall().resolves(workingUploadJob)
+          .onThirdCall().resolves(completeUploadJob);
+      fss.getStatus
+          .onFirstCall().resolves({ status: UploadStatus.WORKING, chunkStatuses: [UploadStatus.WORKING], uploadId: "091234124", chunkSize: 2424 })
+          .onSecondCall().resolves({ status: UploadStatus.COMPLETE, chunkStatuses: [UploadStatus.COMPLETE], uploadId: "091234124", chunkSize: 2424})
+      const fileId = "12343124";
+      const localPath = "/some/path/into/fms/at/test_file.txt";
+      fss.finalize.resolves({
+        errorCount: 0,
+        chunkNumber: 14,
+        uploadId: uploadJob.jobId,
+      });
+      fss.getFileAttributes.resolves({
+        fileId,
+        localPath,
+        name: "",
+        size: 4,
+        md5: "",
+      });
+
+      // Act
+      await fms.upload(uploadJob, noop);
+
+      // Assert
+      expect(fss.getStatus.called).to.be.true;
```

Comment:
What about these assertions tells us a retry for sure happened? If something, could you add a comment saying what it is for future us. If nothing, you can assert what the functions were called with or assert that something wasn't called that is maybe unique to a non-retry code path.

Reply:
commented

Reply:
As you said in the summary, I think this test has some problems.  These assertions strike me as implementation details which could easily change, which makes the test brittle, and it will be confusing to figure out why it broke by a later us.  

Idea: in `retry`, what if a jss job was update with a new `serviceField`, something like `stage: RETRIED`.  The test could then assert that jss was called with exactly that value.

Reply:
Line 252